spec/enumerable_spec.rb

```diff
@@ -0,0 +1,102 @@
+require_relative('../main')
+describe Enumerable do
+  describe '#my_each' do
+    it 'returns enumerator' do
+      expect(%i[1 2 4 5 6].my_each).to(be_an(Enumerator))
+    end
+
+    it 'my_each should return the same as each' do
+      expect(%i[1 2 4 5 6].my_each.to_a).to(eql(%i[1 2 4 5 6].each.to_a))
+    end
+
+    it 'should returns key, value pair' do
+      expect({ name: 'jay', age: 34 }.my_each.to_h).to(eq({ name: 'jay', age: 34 }.each.to_h))
+    end
+
+    it 'returns elements in the range' do
+      expect(%i[0..10].my_each.to_a).to(eq(%i[0..10].each.to_a))
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'returns enumerator' do
+      expect([1, 2, 'ten', 3.5, %w[hello world!]].my_each).to(be_an(Enumerator))
+    end
+
+    it 'should return the indexes' do
+      array = []
+      %w[cat dog fish].my_each_with_index { |_word, idx| array[idx] = idx }
+      expect(array).to(eq([0, 1, 2]))
+    end
```

Comment:
You can compare it with each_with_index by following the example I have given above.